third_party/move/move-vm/types/src/loaded_data/runtime_types.rs

```diff
@@ -194,6 +190,43 @@ pub enum Type {
     U256,
 }
 
+// Cache for the ability of struct. They will be ignored when comparing equality or Ord as they are just used for caching purpose.
+#[derive(Derivative)]
+#[derivative(Debug, Clone, Eq, Hash, PartialEq, Ord, PartialOrd)]
```

Comment:
Seems you needed this for the ignore semantics (shouldn't it have been there already for hashmap?). Not a fan of doing something like this (i.e. cheating on equality :-) ), I think the abilities belong into a different map, but this is for another time.